api/v1alpha1/accessrequest_types.go

```diff
@@ -98,6 +100,31 @@ type AccessRequest struct {
 	Status AccessRequestStatus `json:"status,omitempty"`
 }
 
+// UpdateStatus will update this AccessRequest status field based on
+// the given status and details
+func (ar *AccessRequest) UpdateStatus(status Status, details string) {
```

Comment:
Small thing, feel free to take it or leave it:

In Argo CD, adding logic to `_types.go` files spiraled into a situation where we can't easily implement a go workspace just for types, because we were importing too much stuff.

If you want to get out ahead of that problem, I'd recommend adding an accessrequest_utils.go elsewhere in the project so that this directory is just types.

Reply:
Totally fair and I am well aware of that problem. All the logic in that package should only depend on the local type structs. Personally I would never add additional dependencies to the api package. However I think it is nice to have a bit of OO goodness and allowing the `AccessRequest` object to centralize all the logic to update it's status and not have that spread in different packages. I added a comment in the `godoc` explaining the situation and how to proceed if additional dependencies are required at some point.

Reply:
Sounds good! And I think `time` is a perfectly acceptable import for the api module. Good idea on the godoc.